Packet++/header/TcpLayer.h

```diff
@@ -6,6 +6,7 @@
 #include <string.h>
 
 #ifndef PCPP_DEPRECATED_TCP_OPTION_TYPE
```

Comment:
Maybe we don't need `#ifndef PCPP_DEPRECATED_TCP_OPTION_TYPE`. We can assume no one will define it externally, same as we have with other macros in this project 🤔

Reply:
Well, if we are assuming that then yes, the ifndefs are superfluous.